### PR TITLE
Cargo check on push and pull request (github actions)

### PR DESCRIPTION
**Pull request by vvilhonen**
Tests require bit more work to run on CI but checking compilation is better than nothing

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+  check:
+    name: cargo check
+    runs-on: ubuntu-latest
+    timeout-minutes: 20
+    steps:
+      - uses: actions/checkout@v3
+      - uses: dtolnay/rust-toolchain@master
+        with:
+          toolchain: stable
+      - run: cargo check
```
Comment (1c3t3a) [inline .github/workflows/ci.yml:20]: maybe add `--all-features`?

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,20 @@
+name: CI
+
+on:
+  push:
```
Comment (1c3t3a) [inline .github/workflows/ci.yml:4]: Maybe restrict this only to the `master` branch?

File: .github/workflows/ci.yml (hunk 1 above)
Comment (vvilhonen) [reply .github/workflows/ci.yml:20]: Good idea!

File: .github/workflows/ci.yml (hunk 2 above)
Comment (vvilhonen) [reply .github/workflows/ci.yml:4]: Makes sense, I’ll update this along with the previous comment’s change

**Review by mdrokz — APPROVED** [review]
Looks good thanks for your contribution!

File: .github/workflows/ci.yml (hunk 2 above)
Comment (vvilhonen) [reply .github/workflows/ci.yml:4]: I tried with it and the build wouldn't trigger anymore in my fork's branch after push so there's no way to test changes to gh actions configuration.

**Comment by vvilhonen** [comment]
- Cargo check is now done with `--all-features`
- Warnings turned into errors with `RUSTFLAGS=-Dwarnings`
- Fixed everything the previous changes brought up
- Added format check

**Comment by mdrokz** [comment]
> * Cargo check is now done with `--all-features`
> * Warnings turned into errors with `RUSTFLAGS=-Dwarnings`
> * Fixed everything the previous changes brought up
> * Added format check
> * Build status badge pointing to github actions

great job and thanks for your contribution!

**Comment by mdrokz** [comment]
> * Cargo check is now done with `--all-features`
> * Warnings turned into errors with `RUSTFLAGS=-Dwarnings`
> * Fixed everything the previous changes brought up
> * Added format check
> * Build status badge pointing to github actions

great job and thanks for your contribution!

**Pull request merged**